app/src/main/java/com/cmput301f20t21/bookfriends/ui/request/MapDialog.java

```diff
@@ -0,0 +1,127 @@
+
+    // constructor
+    public MapDialog(RequestViewModel vm, int position) {
+        this.vm = vm;
+        this.position = position;
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        Dialog dialog =  super.onCreateDialog(savedInstanceState);
+        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
+        dialog.setContentView(R.layout.map_dialog);
+
+        MapView mapView = (MapView) dialog.findViewById(R.id.map_view);
+
+        mapView.onCreate(dialog.onSaveInstanceState());
+        mapView.onResume();
+        mapView.getMapAsync(new OnMapReadyCallback() {
+            @Override
+            public void onMapReady(GoogleMap googleMap) {
+                LatLng edmontonLatLng = new LatLng(53.544388, -113.490929);
+                googleMap.moveCamera(CameraUpdateFactory.newLatLngZoom(edmontonLatLng, 12f));
+                myMap = googleMap;
+            }
+        });
+
+        searchText = (EditText) dialog.findViewById(R.id.input_search);
+        searchText.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+            @Override
+            public boolean onEditorAction(TextView textView, int actionId, KeyEvent keyEvent) {
+                if (actionId == EditorInfo.IME_ACTION_SEARCH
+                        || actionId == EditorInfo.IME_ACTION_DONE
+                        || keyEvent.getAction() == keyEvent.ACTION_DOWN
+                        || keyEvent.getAction() == keyEvent.KEYCODE_ENTER) {
+                    // execute searching
+                    String searchString = searchText.getText().toString();
+                    Geocoder geocoder = new Geocoder(dialog.getContext());
+                    List<Address> addressList = new ArrayList<>();
+                    try {
+                        addressList = geocoder.getFromLocationName(searchString, 1);
+                    } catch (IOException e) {
+
+                    }
+
+                    if (addressList.size() > 0) {
+                        // clear the previous marker
+                        myMap.clear();
+                        // get the first address
+                        Address address = addressList.get(0);
+                        LatLng latLng = new LatLng(address.getLatitude(), address.getLongitude());
+                        // move camera to that address
+                        myMap.moveCamera(CameraUpdateFactory.newLatLngZoom(latLng, 15f));
+
+                        MarkerOptions options = new MarkerOptions()
+                                .position(latLng)
+                                .title(address.getAddressLine(0));
+                        myMap.addMarker(options);
+                        // TODO: hide the keyboard after searching (optional)
+                        // not sure why this is not working properly
+                        dialog.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_HIDDEN);
```

Comment:
<details><summary>we could possibly do it this way. works on my end
</summary>

https://stackoverflow.com/a/1109108/7358099
![image](https://user-images.githubusercontent.com/25297856/99490983-2ae4cb00-2928-11eb-83a2-555e9fadbc21.png)
</details>